tensor_genn/algorithms/relu_ann.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+import random
+
+import tensorflow as tf
+from pygenn import genn_model, genn_wrapper
+
+class ReLUANN():
+    def __init__(self,neuron_resting_voltage=-60.0,neuron_threshold_voltage=-55.0,
+        membrane_capacitance=1.0, model_timestep=1.0,):
+        self.Vres = neuron_resting_voltage
+        self.Vthr = neuron_threshold_voltage
+        self.Cm = membrane_capacitance
+        self.timestep = model_timestep
+
+    def convert(self, tf_model):
+        # create custom classes
+        if_model = genn_model.create_custom_neuron_class(
+            "if_model",
+            param_names=["Vres","Vthr","Cm"],
+            var_name_types=[("Vmem","scalar"),("SpikeNumber","unsigned int")],
+            sim_code="""
+            $(Vmem) += $(Isyn)*(DT / $(Cm));
+            //printf("Vmem: %f, Isyn: %f, SpikeNumber: %d", $(Vmem),$(Isyn),$(SpikeNumber));
+            """,
+            reset_code="""
+            $(Vmem) = $(Vres); 
+            $(SpikeNumber) += 1;
+            """,
+            threshold_condition_code="$(Vmem) >= $(Vthr)"
+        )
+
+        cs_model = genn_model.create_custom_current_source_class(
+            "cs_model",
+            var_name_types=[("magnitude","scalar")],
+            injection_code="""
+            $(injectCurrent, $(magnitude));
+            """
+        )
+
+        # Fetch tf_model details
+        n_layers = len(tf_model.layers)
+        tf_weights = tf_model.get_weights()
+
+        # Params and init
+        if_params = {
+            "Vres":self.Vres,
+            "Vthr":self.Vthr,
+            "Cm":self.Cm
+        }
+        if_init = {
+            "Vmem":genn_model.init_var("Uniform", {"min": self.Vres, "max": self.Vthr}),
+            "SpikeNumber":0
+        }
+        
+        cs_init = {"magnitude":10.0}
+
+        # Define model and populations
+        self.g_model = genn_model.GeNNModel("float","g_model")
+        self.neuron_pops = {}
+        self.syn_pops = {}
+
+        for i in range(1,n_layers): # 1,2 - synapses
+            if i == 1:
```

Comment:
It would be good to add checks that the layers are compatible with what this algorithm supports i.e. no bias, dense weights, relu. If not, again, throw ``NotImplementedError``

Reply:
Hi Jamie,
I've implemented the checks for bias and dense layers. But the tf.keras API doesn't seem to have an activation function attribute that I can access. Any ideas on a workaround solution?
Thanks
Ajay